src/utils/keystore.ts

```diff
@@ -186,25 +247,29 @@ export async function recover (password, keyObject) {
  * @return {Object}
  */
 export async function dump (
-  name,
-  password,
-  privateKey,
-  nonce = nacl.randomBytes(24),
-  salt = nacl.randomBytes(16),
-  options = {}
+  name: string,
+  password: string,
+  privateKey: string|Buffer,
+  nonce: Buffer|Uint8Array = nacl.randomBytes(24),
+  salt: Buffer|Uint8Array = nacl.randomBytes(16),
+  options:{
+    kdf?: string,
+    cipher?:string,
+    kdfparams?: Kdf_params
+  } = {}
 ) {
   const opt = Object.assign({}, DEFAULTS.crypto, options)
   return marshal(
     name,
     await deriveKey(password, salt, opt),
     typeof privateKey === 'string' ? Buffer.from(privateKey, 'hex') : privateKey,
-    nonce,
-    salt,
+    nonce as Buffer,
+    salt as Buffer,
     opt
   )
 }
 
-export function validateKeyObj (obj) {
+export function validateKeyObj (obj: any) {
```

Comment:
```suggestion
export function validateKeyObj (obj: Partial<SecretStorageKeyObject>) {
```

see previous comment